src/components/CommandPalette/CommandPalette.js

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useRef } from 'react'
+import { Modal } from '../Modal'
+import { IconSearch } from '../Icon/icons/IconSearch'
+import { IconX } from '../Icon/icons/IconX'
+import CommandPaletteResults from './CommandPaletteResults'
+import { action } from '@storybook/addon-actions'
+
+const NoResults = ({ term }) => (
+  <div className="flex items-center justify-center min-h-[200px] dark:text-dark-200 opacity-50 bg-gray-200">
+    No results found for <strong>{term}</strong>
+  </div>
+)
+
+export default function Default({
+  onToggle,
+  visible,
+  onCancel,
+  results,
+  term,
+  onSearch,
+  loading
+}) {
+  let keysPressed = {}
+  const resultsRef = useRef([])
+  const inputRef = useRef()
+
+  useEffect(() => {
+    document.addEventListener('keydown', handleKeyDown)
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown)
+    }
+  })
+
+  const handleKeyDown = (e) => {
+    if ((e.keyCode === 75 && e.metaKey) || (e.ctrlKey && e.keyCode === 75)) {
+      onToggle()
+    }
+  }
+
+  const handleShortcuts = (e) => {
+    keysPressed[e.key] = true
+    if (e.key === 'Escape') {
+      e.preventDefault()
+      onCancel()
+    }
+
+    if (results?.list?.length) {
+      if (e.keyCode === 40) {
+        // Down arrow
+        e.preventDefault()
+        resultsRef.current[1].focus()
+      }
+
+      if (e.keyCode === 38) {
+        // Up arrow
+        e.preventDefault()
+        resultsRef.current[results?.list?.length].focus()
+      }
+    }
+  }
+
+  return (
+    <Modal
+      visible={visible}
+      onCancel={onCancel}
+      hideFooter={true}
+      contentStyle={{ border: 0 }}
+    >
+      <div className="rounded-lg overflow-hidden">
+        <div
+          className={`flex items-center w-full bg-white rounded-t-lg dark:bg-scale-500`}
+        >
+          <div className="px-2">
+            <IconSearch size={18} stroke="#999" strokeWidth="1.8" />
+          </div>
+          <input
+            ref={(el) => (inputRef.current = el)}
+            onKeyDown={(e) => handleShortcuts(e)}
+            value={term}
+            onChange={onSearch}
+            className="text-sm md:text-base pr-3 py-3 w-full focus:outline-none bg-transparent placeholder-dark-300 dark:placeholder-dark-400 dark:text-white"
```

Comment:
@rishimohan You're probably going to have to move all the CSS classes into the theme file.

Reply:
using this defaultTheme

https://github.com/supabase/ui/blob/alpha/src/lib/theme/defaultTheme.ts

Reply:
then import classes like this
https://github.com/supabase/ui/blob/87776677af000bed91998403b8d3f936b8b10592/src/components/Button/Button.tsx#L88